### DISCUSSION TITLE
Caching the output of ImageFromDockerfileBuilder

### DISCUSSION THREAD

**Discussion by PappBenedek** [testcontainers/testcontainers-dotnet#1036] [Q&A] [answered]
Hey folks,
My image build time is quite long, I'm wondering is there a way of caching this? 
EDIT: the CreateAsync() on the FutureImage is the annoyingly long part
I did not find anything about this in documentation.
Thanks in advance

**Comment by HofmeisterAn** [comment]
Docker's BuildKit is not available through the API (AFAIK, not much we can do here - sorry). I recommend using the Docker CLI instead to build the image.

**Reply by PappBenedek** [reply]
got you, but is there any way that i can use the result of that within the testcontainers?

**Reply by HofmeisterAn** [reply, marked as answer]
Locally or on CI? You can use runsettings or environment variables to pass the actual image to the test. Or an Implementation of `IImage` that resolves the previously built image.

**Reply by PappBenedek** [reply]
Implementing the IImage resolves the problem, really appreciate your input. Thanks!

**Reply by HofmeisterAn** [reply]
That's great! If you'd like, you can share it here, and others can use or build upon it (I'm definitely interested in). I understand that these suggestions may not be very elegant, but I couldn't think of any better ideas quickly. Perhaps someone will come up with something better.

You can also run the Docker build command as a subprocess from your tests. This way, you'll have access to the chosen tag without needing to pass it into your tests.

Another idea that crossed my mind is using labels to choose images built by another process. Testcontainers could select an image based on its labels (although this feature is not yet implemented).
